Some changes to make this crate usable with a stable toolchain.

egrimley-arm opened the pull request:
- Remove the "nightly" feature from dependencies.hashbrown.
- Remove the #![feature(alloc_prelude))].
- Remove pub use alloc::prelude::v1::*;
- Add some "use alloc::..." lines to compensate.

There have been no changes upstream since 2020-03-05.

egrimley-arm commented:
There's a possible problem here with `use-std` and `alloc`.

nspin commented:
Just to clarify, is it the case that upstream pinecone is incompatible with the stable channel of Rustc?

egrimley-arm commented:
> Just to clarify, is it the case that upstream pinecone is incompatible with the stable channel of Rustc?

Yes, that's my understanding.

There's a bit missing in this version of the PR. I'm about to push an update ...

egrimley-arm committed:
Some changes to make this crate usable with a stable toolchain.

- Remove the "nightly" feature from dependencies.hashbrown.
- Remove the #![feature(alloc_prelude))].
- Remove pub use alloc::prelude::v1::*;
- Add some (conditional) "use alloc::..." lines to compensate.

There have been no changes upstream since 2020-03-05.

mathias-arm commented:
Is that the only crate preventing to use a stable compiler?

egrimley-arm commented:
> Is that the only crate preventing to use a stable compiler?

No, there seem to be three: `pinecone`, `wasi-types`, `wasmtime`. There are also PRs for the other two, which were already forked by `veracruz-project`, if you want to take a look.

mathias-arm commented:
Looking at the description of `pinecone`, I am wondering if we should not switch to `postcard` (`pinecone` is a fork of `postcard`) or another binary serialization crate (we also use `bincode`).

nspin commented:
IceCap itself uses pinecone internally because it doesn't depend on std. We would still need to patch pinecone if we want to build IceCap with the stable channel.

mathias-arm commented:
`postcard` from which `pinecone` derives is also `no_std` and does not require `alloc` (`alloc` is optional).

nspin commented:
Oops - I was under the impression that pinecone was postcard minus std, but it appears I'm mistaken. Then I can look into replacing pinecone with postcard throughout IceCap.
